python/golem_task_api/testutils/tasklifecycleutil.py

```diff
@@ -105,54 +113,42 @@ def get_task_api_service(work_dir: Path):
 
             assert not await self.requestor_client.has_pending_subtasks(task_id)
+            return self.req_dir.task_dir(task_id)
 
-    def mkdir_requestor(self, task_id: str) -> None:
-        self.req_task_work_dir = self.req_work_dir / task_id
-        self.req_task_work_dir.mkdir()
-        self.req_task_inputs_dir = \
-            self.req_task_work_dir / constants.TASK_INPUTS_DIR
-        self.req_task_inputs_dir.mkdir()
-        self.req_subtask_inputs_dir = \
-            self.req_task_work_dir / constants.SUBTASK_INPUTS_DIR
-        self.req_subtask_inputs_dir.mkdir()
-        self.req_task_outputs_dir = \
-            self.req_task_work_dir / constants.TASK_OUTPUTS_DIR
-        self.req_task_outputs_dir.mkdir()
-        self.req_subtask_outputs_dir = \
-            self.req_task_work_dir / constants.SUBTASK_OUTPUTS_DIR
-        self.req_subtask_outputs_dir.mkdir()
-
-    def mkdir_provider_task(self, task_id: str) -> None:
-        self.prov_task_work_dir = self.prov_work_dir / task_id
-        self.prov_task_work_dir.mkdir()
-        self.prov_subtask_inputs_dir = \
-            self.prov_task_work_dir / constants.SUBTASK_INPUTS_DIR
-        self.prov_subtask_inputs_dir.mkdir()
-
-    def mkdir_provider_subtask(self, subtask_id: str) -> None:
-        prov_subtask_work_dir = self.prov_task_work_dir / subtask_id
-        prov_subtask_work_dir.mkdir()
-
-    def put_resources_to_requestor(self, resources: List[Path]) -> None:
+    def put_resources_to_requestor(
+            self,
+            resources: List[Path],
+            task_id: str
+    ) -> None:
+        task_inputs_dir = self.req_dir.task_dir(task_id).task_inputs_dir
         for resource in resources:
-            shutil.copy2(resource, self.req_task_inputs_dir)
+            shutil.copy2(resource, task_inputs_dir)
 
-    def copy_resources_from_requestor(self, resources: List[str]) -> None:
+    def copy_resources_from_requestor(
+            self,
+            resources: List[str],
+            task_id: str
+    ) -> None:
+        req_dir = self.req_dir.task_dir(task_id).subtask_inputs_dir
+        prov_dir = self.prov_dir.task_dir(task_id).subtask_inputs_dir
         for resource_id in resources:
-            network_resource = self.req_subtask_inputs_dir / resource_id
+            network_resource = req_dir / resource_id
             assert network_resource.exists()
-            shutil.copy2(network_resource, self.prov_subtask_inputs_dir)
+            shutil.copy2(network_resource, prov_dir)
 
     def copy_result_from_provider(
             self,
             output_filepath: Path,
-            subtask_id: str,
+            task_id: str,
+            subtask_id: str
     ) -> None:
-        result = self.prov_task_work_dir / output_filepath
+        result = self.prov_dir.task_dir(task_id) / output_filepath
+        outputs_dir = self.req_dir.task_dir(task_id).subtask_outputs_dir
+        # TODO: Shouldn't it be subtask_outputs_dir / subtask_id ??
```

Comment:
I think yes, checking the current RTM version it also misses the subtask_id, is there an issue for this?

Reply:
lets make an issue and fix it :D

Reply:
I'll do this as a part of this task because releasing a new task api version and then updating both blenderapp and golem seems too much hassle over such a tiny fix.

Reply:
Fixed. https://github.com/golemfactory/task-api/compare/0c1dfbf6a126177687845e49549ff867c2b2b476..bfadbc9359302ca0b1ef36f91f342382e4f26355